app/models/user.rb

```diff
@@ -9,4 +9,55 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_arrayb = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.concat(friends_arrayb)
+    friends_array.compact
+  end
+
+  def send_invitation(user_id)
+    @friendship = Friendship.new(user_id: id, friend_id: user_id)
+    @friendship.confirmed = false
+    @friendship.save
+  end
+
+  # Users who have yet to confirme friend requests
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  end
+
+  # Users who have requested to be friends
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  end
+
+  def confirm_friend(user)
+    friendship = inverse_friendships.where(user_id: user).first
+    friendship.confirmed = true
+    friendship.save
+  end
+
+  def friend_invites(user_id)
+    friendship = friendships.where(friend_id: user_id).first
+    true if friendship && friendship.confirmed == false
+  end
+
+  def receive_invitation(user_id)
+    friendship = inverse_friendships.where(user_id: user_id).first
+    true if friendship && friendship.confirmed == false
+  end
```

Comment:
**Optional** If you think a little more, you can avoid creating these methods and use associations instead. That's the power of Rails' associations!